pkg/csi/service/vanilla/controller_helper.go

```diff
@@ -192,15 +189,13 @@ func getBlockVolumeIDToNodeUUIDMap(ctx context.Context, c *controller,
 	// Iterate through all the VMs and build the vmMoIDToHostUUID map
 	// and the volumeID to VMMoiD map
 	for _, info := range vmMoList {
-		vmMoID := info.Reference().Value
-
 		devices := info.Config.Hardware.Device
 		vmDevices := object.VirtualDeviceList(devices)
 		for _, device := range vmDevices {
 			if vmDevices.TypeName(device) == "VirtualDisk" {
 				if virtualDisk, ok := device.(*types.VirtualDisk); ok {
 					if virtualDisk.VDiskId != nil {
-						volumeIDNodeUUIDMap[virtualDisk.VDiskId.Id] = vmRefToUUID[vmMoID]
+						volumeIDNodeUUIDMap[virtualDisk.VDiskId.Id] = info.Config.Uuid
```

Comment:
I vaguely remember trying to use info.Config.uuid but it had given empty or a different value. Can you please check if nodeVM.UUID and info.Config.Uuid is the same?

Reply:
I meant, if you could print and check the value vmRefToUUID[vmMoID] and info.Config.Uuid are the same, and paste the logs in the description that would be great.